slatedb/src/mem_table_flush.rs

```diff
@@ -181,141 +186,65 @@ impl MemtableFlusher {
         }
         Ok(())
     }
-}
 
-impl DbInner {
-    async fn flush_and_record(
-        self: &Arc<Self>,
-        flusher: &mut MemtableFlusher,
-    ) -> Result<(), SlateDBError> {
-        let result = flusher.flush_imm_memtables_to_l0().await;
+    async fn flush_and_record(&mut self) -> Result<(), SlateDBError> {
+        let result = self.flush_imm_memtables_to_l0().await;
         if let Err(err) = &result {
             error!("error from memtable flush [error={:?}]", err);
         } else {
-            self.db_stats.immutable_memtable_flushes.inc();
+            self.db_inner.db_stats.immutable_memtable_flushes.inc();
         }
         result
     }
+}
 
-    pub(crate) fn spawn_memtable_flush_task(
-        self: &Arc<Self>,
-        manifest: FenceableManifest,
-        mut flush_rx: UnboundedReceiver<MemtableFlushMsg>,
-        tokio_handle: &Handle,
-    ) -> Option<tokio::task::JoinHandle<Result<(), SlateDBError>>> {
-        let this = Arc::clone(self);
-
-        async fn core_flush_loop(
-            this: &Arc<DbInner>,
-            flusher: &mut MemtableFlusher,
-            flush_rx: &mut UnboundedReceiver<MemtableFlushMsg>,
-        ) -> Result<(), SlateDBError> {
-            let mut manifest_poll_interval = this
-                .system_clock
-                .ticker(this.settings.manifest_poll_interval);
-            let mut err_reader = this.state.read().error_reader();
+#[async_trait]
+impl MessageHandler<MemtableFlushMsg> for MemtableFlusher {
+    fn tickers(&mut self) -> Vec<(Duration, Box<MessageFactory<MemtableFlushMsg>>)> {
+        vec![(
+            self.db_inner.settings.manifest_poll_interval,
+            Box::new(move || MemtableFlushMsg::PollManifest),
```

Comment:
I should probably understand rust memory management better, but I'm curious why we need the move here?